plutus-benchmark/bls-benchmarks/src/PlutusBenchmark/BLS12_381/Common.hs

```diff
@@ -0,0 +1,349 @@
+                          , 0xa8, 0xc5, 0xd9, 0x85, 0xb3, 0x41, 0x1f, 0xc5
+                          , 0xfd, 0x44, 0xaf, 0x60, 0x7e, 0x42, 0x92, 0x3e
+                          , 0xab, 0xb4, 0x7a, 0xd8, 0x76, 0xe1, 0xf0, 0x2b
+                          , 0x5b, 0xe0, 0x34, 0xad, 0xaf, 0x73, 0x95, 0x2a
+                          , 0xe8, 0xaf, 0xfe, 0xe5, 0xf5, 0x18, 0x41, 0xde ]
+
+a :: SerialisedG1Element
+a = mkG1Element [ 0xa0, 0x5b, 0xe5, 0x0f, 0xab, 0x57, 0x95, 0xbb
+                , 0x87, 0x84, 0x39, 0x3a, 0x50, 0x45, 0xf9, 0x87
+                , 0x47, 0x17, 0x3a, 0xd2, 0x87, 0xf5, 0x5e, 0x21
+                , 0x34, 0x71, 0xbd, 0x55, 0x97, 0x45, 0x55, 0x14
+                , 0x52, 0x45, 0x3c, 0x4c, 0x3a, 0x39, 0xe7, 0xc8
+                , 0x83, 0x10, 0x84, 0x9f, 0x3c, 0x7a, 0x1f, 0xc3 ]
+
+b :: SerialisedG2Element
+b = mkG2Element [ 0xad, 0x63, 0x48, 0xb6, 0xb7, 0xb3, 0x4c, 0x86
+                , 0xbf, 0x37, 0xa7, 0x48, 0xcd, 0x2d, 0x82, 0xa2
+                , 0x50, 0xdf, 0xc6, 0x48, 0x46, 0x75, 0x66, 0x88
+                , 0x25, 0xa1, 0x6f, 0x7d, 0xa6, 0xa0, 0x4d, 0x34
+                , 0x24, 0x11, 0x3e, 0x32, 0x5c, 0xe7, 0x34, 0xec
+                , 0x44, 0x95, 0x60, 0x82, 0xc0, 0xa0, 0x6e, 0x5f
+                , 0x18, 0x68, 0xe1, 0xf1, 0xa6, 0xe5, 0x59, 0xb9
+                , 0xfe, 0x81, 0xf1, 0xa9, 0x01, 0xf8, 0xa6, 0x34
+                , 0x1b, 0x30, 0x1c, 0x45, 0xb2, 0x5d, 0x30, 0x80
+                , 0xfb, 0xc5, 0x03, 0x93, 0x53, 0xd8, 0xf7, 0x1b
+                , 0x55, 0x0b, 0x27, 0x4e, 0xc4, 0xc0, 0x7c, 0x70
+                , 0xcd, 0x11, 0x53, 0x56, 0x2c, 0x31, 0x4c, 0x97 ]
+
+c :: SerialisedG1Element
+c = mkG1Element [ 0xb5, 0x69, 0xcc, 0x49, 0x1b, 0x4d, 0xf0, 0x35
+                , 0xcb, 0xf4, 0x9e, 0x95, 0x1f, 0xd4, 0xfe, 0x30
+                , 0xaa, 0x82, 0x36, 0xb0, 0xe2, 0xaf, 0x68, 0xf4
+                , 0xc1, 0x59, 0x2c, 0xd4, 0x0d, 0xeb, 0xeb, 0x71
+                , 0x8a, 0xf3, 0x36, 0x39, 0xdb, 0x6b, 0xc1, 0xe2
+                , 0xda, 0x9d, 0x98, 0xe5, 0x53, 0xe5, 0xea, 0xed ]
+
+{-# INLINABLE groth16Verify #-}
+groth16Verify
+    :: BuiltinByteString  -- G1
+    -> BuiltinByteString  -- G2
+    -> BuiltinByteString  -- G2
+    -> BuiltinByteString  -- G1
+    -> BuiltinByteString  -- G1
+    -> BuiltinByteString  -- G1
+    -> BuiltinByteString  -- G1
+    -> BuiltinByteString  -- G2
+    -> BuiltinByteString  -- G1
+    -> Integer
+    -> Bool
+groth16Verify alpha' beta' gamma' delta' abc1' abc2' a' b' c' s =
+    let alpha = Tx.bls12_381_G1_uncompress alpha'
+        beta  = Tx.bls12_381_G2_uncompress beta'
+        gamma = Tx.bls12_381_G2_uncompress gamma'
+        delta = Tx.bls12_381_G2_uncompress delta'
+        abc1  = Tx.bls12_381_G1_uncompress abc1'
+        abc2  = Tx.bls12_381_G1_uncompress abc2'
+        a     = Tx.bls12_381_G1_uncompress a'
+        b     = Tx.bls12_381_G2_uncompress b'
+        c     = Tx.bls12_381_G1_uncompress c'
```

Comment:
Not strictly necessary but this can be a good time for `ViewPatterns`
```
groth16Verify (Tx.bls12_381_G1_uncompress -> alpha) ...
```

Reply:
I did that.  I'm not sure whether it's better, but it's certainly _different_.